### PR TITLE
Prepare CI for releasing the website

### PR DESCRIPTION
**Pull request by cfcosta**
Needs major testing, but it works for me.

### CODE REVIEW

**Comment by cfcosta** [comment]
Depends on https://github.com/kiali/kiali.io/pull/247

**Comment by jshaughn** [comment]
@cfcosta I defer to @israel-hdez for this review.  I really don't know anything about the jenkins automation.  My only question is whether there is github issue for this work, so that we know what the PR is actually supposed to accomplish.

**Comment by cfcosta** [comment]
Closes #2181.

File: deploy/jenkins-ci/Jenkinsfile [hunk 1]
```diff
@@ -126,6 +139,27 @@ node('kiali-build') {
...
+            submoduleCfg: [],
+            userRemoteConfigs: [[
+            credentialsId: 'kiali-bot-gh-ssh',
+            url: params.SITE_GITHUB_URI]]
+        ])
+
+          sh "cd ${siteDir}; git config user.email 'kiali-dev@googlegroups.com'"
+          sh "cd ${siteDir}; git config user.name 'kiali-bot'"
+          sh "ls src"
+          sh "ls src/github.com"
```
Comment (israel-hdez) [inline deploy/jenkins-ci/Jenkinsfile:160]: Perhaps remove these `ls` commands that won't be useful for the final code?

File: deploy/jenkins-ci/Makefile.site.jenkins [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+website-push-version-tag:
+ifeq ($(PUSH_CHANGES),y)
+	git push $(SITE_GITHUB_URI) $$(git rev-parse HEAD):refs/tags/v$(SITE_VERSION)
+ifdef GH_TOKEN
+	@echo "Creating GitHub release..."
+	@curl -H "Authorization: token $(GH_TOKEN)" \
+	  -H "Content-Type: application/json" \
+	  -d '{"name": "Kiali.io $(SITE_VERSION)", "tag_name": "$(VERSION_TAG)"}' \
+	  -X POST $(SITE_RELEASE_URI)
+endif
```
Comment (israel-hdez) [inline deploy/jenkins-ci/Makefile.site.jenkins:55]: I think we don't need to create releases for kiali.io repository
It's OK to create them, but I'd prefer to not :)

File: deploy/jenkins-ci/Makefile.site.jenkins [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+.PHONY: website-build-archive website-push-version-tag
+.PHONY: all release
+
+all:
+	$(error You must explicitly specify a target)
+
+website-build-archive:
+ifneq ($(PUSH_CHANGES),y)
```
Comment (israel-hdez) [inline deploy/jenkins-ci/Makefile.site.jenkins:34]: isn't this conditional inverted?

File: deploy/jenkins-ci/Jenkinsfile [hunk 4]
```diff
@@ -52,6 +61,9 @@
  *   - SKIP_OPERATOR_RELEASE
  *      defaultValue: n
  *      description: Set to 'y' if you don't want to release the operator
+ *   - SKIP_SITE_RELEASE
+ *      defaultValue: n
+ *      description: Set to 'y' if you don't want to release the website
```
[the remark this answers is not in the document]
Comment (cfcosta) [reply deploy/jenkins-ci/Jenkinsfile:66]: The idea was to allow the user to decide to even run or not the website release, for tests and so on.

Glad to remove if you see no value in having this.

**Comment by cfcosta** [comment]
> As a side comment, this is becoming too big IMO and starts to become a little hard to maintain. I think we should break this up in three parts: kiali app (back and front-ends), operator and website parts. I'll create issues for breaking this in parts hoping that this can be prioritized... cc @jshaughn .

Agree, that could be the next step (and I think we could also simplify the release process itself, now that operator is its own thing, we can make things simpler).

File: deploy/jenkins-ci/Jenkinsfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (israel-hdez) [reply deploy/jenkins-ci/Jenkinsfile:66]: I was thinking this should be a variable accepting 3 piossible values?
* `auto` (the default) to do the current behavior
* `y` to force releasing the site release regardless of the RELEASE_TYPE
* `n` to force skipping the site release regardless of the RELEASE_TYPE

**Comment by jshaughn** [comment]
> As a side comment, this is becoming too big... cc @jshaughn

I trust you guys to do the right thing here.  Proceed any way you both agree on.

**Comment by israel-hdez** [comment]
I created the issue for splitting this big pipeline in smaller parts: https://github.com/kiali/kiali/issues/2895

Let's move forward this PR as is and work later on splitting/simplifying.

**Comment by cfcosta** [comment]
@israel-hdez as it turns out, if you run two different tasks from a Makefile, it gets evaluated twice. The version is bumped on the first release, and then re-evaluated when pushing the tags. Should be ok now.

**Pull request merged**